### PR TITLE
*: trace statsCache and preparePlanCache by Global memory tracker.

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed: Polish the implement of Global Memory Limit
1. Attach all statsCache tracker to GlobalLRUMemUsageTracker, to trace stats cache memory usage.
2. Re-implement the way to trace the memory usage for PreparePlanCache.
3. Add an attribute `ignoreAction` in Tracker and use it for all cache, to avoid potential tidb panic. 

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

File: util/profile/trackerRecorder.go [hunk 1]
```diff
@@ -31,24 +31,38 @@ func HeapProfileForGlobalMemTracker(d time.Duration) {
...
+)
+
+func heapProfileForPreparePlanCache() error {
+	total := int64(0)
+	bytes, err := col.getFuncMemUsage(PreparePlanCacheKey)
+	if err != nil {
+		return err
+	}
+	total += bytes
+	bytes, err = col.getFuncMemUsage(PreparePlanCacheValue)
```
Comment (Yisaer) [inline util/profile/trackerRecorder.go:55]: `getFuncMemUsage` will  parse the memory heap and travel the heap structure tree each time. I think we can leave a `TODO` here to improve `getFuncMemUsage` by receiving multiple function keys.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM. plz add a test for `heapProfileForPreparePlanCache`

File: util/profile/trackerRecorder.go (hunk 1 above)
Comment (wshwsh12) [reply util/profile/trackerRecorder.go]: Addressed.

**Review by Yisaer — COMMENTED** [review]
LGTM

**Comment by zz-jason** [comment]
I'm going to close this PR since it hasn't been updated for a long time. feel free to reopen if you want to continue with it. thank you for your contribution.

**Pull request closed without merging**